chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/UserTestingLabMapDAOImpl.java

```diff
@@ -0,0 +1,163 @@
+        return new UserTestingLabMapDTO(entity);
+    }
+
+    public void delete(UserTestingLabMapDTO dto) throws EntityRetrievalException {
+        update(getUserTestingLabMapEntity(dto));
+    }
+
+    @Override
+    public List<UserTestingLabMapDTO> getByUserId(Long userId) {
+        Query query = entityManager
+                .createQuery(
+                        "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab tl "
+                                + "join fetch utlm.user u " + "where (utlm.deleted != true) AND (u.id = :userId) ",
+                        UserTestingLabMapEntity.class);
+        query.setParameter("userId", userId);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        List<UserTestingLabMapDTO> dtos = new ArrayList<UserTestingLabMapDTO>();
+        if (result != null) {
+            for (UserTestingLabMapEntity entity : result) {
+                dtos.add(new UserTestingLabMapDTO(entity));
+            }
+        }
+        return dtos;
+    }
+
+    @Override
+    public List<UserTestingLabMapDTO> getByAtlId(Long atlId) {
+        Query query = entityManager.createQuery(
+                "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab tl "
+                        + "join fetch utlm.user u where (utlm.deleted != true) AND (tl.id = :atlId) ",
+                UserTestingLabMapEntity.class);
+        query.setParameter("atlId", atlId);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        List<UserTestingLabMapDTO> dtos = new ArrayList<UserTestingLabMapDTO>();
+        for (UserTestingLabMapEntity entity : result) {
+            dtos.add(new UserTestingLabMapDTO(entity));
+        }
+        return dtos;
+    }
+
+    @Override
+    public UserTestingLabMapDTO getById(Long id) {
+        Query query = entityManager.createQuery(
+                "from UserTestingLabMapEntity utlm " + "join fetch utlm.testingLab TestingLabEntity "
+                        + "join fetch utlm.user UserEntity" + "where (utlm.deleted != true) AND (utlm.id = :id) ",
+                UserTestingLabMapEntity.class);
+        query.setParameter("id", id);
+        List<UserTestingLabMapEntity> result = query.getResultList();
+
+        if (result.size() == 0) {
+            return null;
+        }
+        return new UserTestingLabMapDTO(result.get(0));
+    }
+
+    private UserTestingLabMapEntity create(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
+        if (entity.getRetired() == null) {
+            entity.setRetired(false);
+        }
+        if (entity.getDeleted() == null) {
+            entity.setDeleted(false);
+        }
+        entityManager.persist(entity);
+        return entity;
+    }
+
+    private UserTestingLabMapEntity update(UserTestingLabMapEntity entity) {
+        entity.setLastModifiedUser(getUserId(User.SYSTEM_USER_ID));
```

Comment:
Should this be the system user id? Is it possible there is an actual logged in user updating this relationship and their ID should be here but default to the system id if null?

Reply:
This function gets the current user, but if there isn't a current for some reason (maybe running in the context of a Quartz job running), it will return `User.SYSTEM_USER_ID` .